src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/TargetRegisterMap.cs

```diff
@@ -22,6 +22,7 @@ public TargetRegisterMap(TargetOS os)
             switch (os)
             {
                 case TargetOS.Windows:
+                case TargetOS.Linux:
```

Comment:
ARM calling convention is same on all OSes. It would be better to delete this switch.